tests/e2e/fixtures/kserve_denpendencies.py

```diff
@@ -0,0 +1,196 @@
+import os
+import logging
+import re
+import subprocess
+import time
+import pytest
+from e2e.utils.constants import TENSORFLOW_SERVING_VERSION
+from e2e.utils.s3_for_kserve.data_bucket import S3BucketForKServe
+from e2e.fixtures.cluster import (
+    cluster,
+    create_iam_service_account,
+    associate_iam_oidc_provider,
+    delete_iam_service_account,
+)
+from e2e.utils.utils import (
+    rand_name,
+    write_yaml_file,
+    exec_shell,
+    wait_for,
+    load_yaml_file,
+    kubectl_apply,
+)
+from e2e.utils.config import configure_resource_fixture, metadata
+from e2e.conftest import region
+from e2e.utils.load_balancer.setup_load_balancer import create_certificates
+from e2e.utils.custom_resources import get_inference_service
+
+
+RANDOM_PREFIX = rand_name("kserve-")
+SECRET_CONFIG_FILE = "./resources/kserve/kserve-secret.yaml"
+INFERENCE_CONFIG_FILE = "./resources/kserve/inference-service.yaml"
+AUTHORIZATION_POLICY_CONFIG_FILE = "./resources/kserve/allow-predictor.yaml"
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope="class")
+def clone_tensorflow_serving():
+    tensorflow_serving_path = "../../serving"
+    if not os.path.isdir(tensorflow_serving_path):
+        retcode = subprocess.call(
+            f"git clone --branch {TENSORFLOW_SERVING_VERSION} https://github.com/tensorflow/serving/ ../../serving".split()
+        )
+        assert retcode == 0
+    else:
+        print("serving directory already exists, skipping clone ...")
+
+
+@pytest.fixture(scope="class")
+def associate_oidc(cluster, region):
+    associate_iam_oidc_provider(cluster_name=cluster, region=region)
+
+
+@pytest.fixture(scope="class")
+def kserve_iam_service_account(metadata, cluster, region, request):
+    metadata_key = "aws-sa"
+
+    service_account_name = "aws-sa"
+
+    def on_create():
+        create_iam_service_account(
+            service_account_name=service_account_name,
+            namespace="kubeflow-example-com",
+            cluster_name=cluster,
+            region=region,
+            iam_policy_arns=["arn:aws:iam::aws:policy/AmazonS3FullAccess"],
+        )
+
+    def on_delete():
+        delete_iam_service_account(
+            service_account_name=service_account_name,
+            namespace="kubeflow-example-com",
+            cluster_name=cluster,
+            region=region,
+        )
+
+    return configure_resource_fixture(
+        metadata=metadata,
+        request=request,
+        resource_details="created",
+        metadata_key=metadata_key,
+        on_create=on_create,
+        on_delete=on_delete,
+    )
+
+
+@pytest.fixture(scope="class")
+def s3_bucket_with_training_data(metadata, cluster, region, request):
```

Comment:
we have this fixture for S3 bucket, do we need `S3BucketforKServe` class in `tests/e2e/utils/s3_for_kserve/data_bucket.py` ? seems unnecessary unless the class is generic to create S3 bucket and delete it and test specific things like data copy is handled here